src/components/dropdown.component.js

```diff
@@ -4,15 +4,51 @@ export default class DropdownComponent {
     constructor(query) {
         this.query = query;
 
+        this.dropdownContainer = $(`${query} .dropdown-container`);
         this.btn = $(`${query} .dropdown-btn`);
-        this.items = $(`${query} .dropdown-item`);
         this.list = $(`${query} .dropdown-list`);
+        this.items = $(`${query} .dropdown-item`);
         this.selectedItem = $(`${query} .selected`);
 
+        this.enabled = true;
+
         this.addDropdownClickEventListner();
         this.addItemClickEventListner();
     }
 
+    getSelectedItem() {
+        return this.selectedItem.text();
+    }
+
+    checkQuery() {
+        if (
+            this.query.length === 0 ||
+            this.dropdownContainer.length === 0 ||
+            this.btn.length === 0 ||
+            this.list.length === 0 ||
+            this.items.length === 0
+        ) {
+            return false;
+        }
+        return true;
+    }
+
+    setEnabled() {
+        this.enabled = this.dropdownContainer.prop('disabled', false);
+        this.dropdownContainer.removeClass('disabled');
```

Comment:
Using classes for this operation is not a good solution because we create a dependency on an external element. The class must exist to be effective. If the class exists but doesn't do what you want?

Reply:
Ok, now i do add "this.dropdown.container.css('pointer-events', 'none');" to the disable method and "this.dropdown.container.css('pointer-events', 'auto');" to enable method, like this i can still remove pointer events from the entire element and keep independency.